### PR TITLE
 Feat: added support to rancher v2.2.x

### PR DESCRIPTION
**Pull request by rawmind0**
This PR constains:
- Updated vendor files.
- Updated go files for already existing resources.
- Updated doc files for already existing resources.
- Added new resource `cloud-credential` to support cloud credential to node dirvers.
- Added new resource `etcd-backup` to support etcd backups with local and S3 storage backends. 
- Added new resource `cluster-driver` to support kontainer engine drivers.
- Added support on rke clusters to:
  - `etcd-backup` with local and S3 storage backends on rke clusters.
  - `dns` configs 
  - `weave` network provider
- Fix for moving namespace to a project after import it from k8s.

Fix issues #8, #65, #73, #78, #79 and #82

### CODE REVIEW

File: website/docs/r/etcdBackup.html.markdown [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+resource "rancher2_etcd_backup" "foo" {
+  backup_config {
+    enabled = true
+    interval_hours = 20
+    retention = 10
+    s3_backup_config {
+      access_key = "access_key"
+      bucket_name = "bucket_name"
+      endpoint = "endpoint"
+      region = "region"
```
Comment (cloudnautique) [inline website/docs/r/etcdBackup.html.markdown]: add `secret_key`

Comment (rawmind0) [reply website/docs/r/etcdBackup.html.markdown]: added

**Comment by drpebcak** [comment]
@rawmind0 do we need to require `access key` and `secret key` for aws? It would be nice if we could just use an instance profile instead.

**Comment by rawmind0** [comment]
@drpebcak , do you mean if we need to require `access key` and `secret key` for aws `cloud-credential`?? Yes, we need them, btm aws instance profile is not supported by the API.

**Pull request merged**